airflow/decorators/base.py

```diff
@@ -337,30 +361,47 @@ def _merge_kwargs(
     return {**kwargs1, **kwargs2}
 
 
-@attr.define(kw_only=True)
+@attr.define(kw_only=True, repr=False)
 class DecoratedMappedOperator(MappedOperator):
     """MappedOperator implementation for @task-decorated task function."""
 
     multiple_outputs: bool
     python_callable: Callable
 
-    def create_unmapped_operator(self, dag: "DAG") -> BaseOperator:
+    # We can't save these in partial_kwargs because op_args and op_kwargs need
+    # to be present in mapped_kwargs, and MappedOperator prevents duplication.
+    partial_op_kwargs: Dict[str, Any]
+
+    @classmethod
+    @cache
+    def get_serialized_fields(cls):
+        return MappedOperator.get_serialized_fields() | {"partial_op_kwargs"}
```

Comment:
Any reason to not user super here?

```suggestion
        return super().get_serialized_fields() | {"partial_op_kwargs"}
```

Reply:
Somehow `super()` doesn’t work here:

```pycon
>>> from airflow.decorators.base import DecoratedMappedOperator
>>> DecoratedMappedOperator.get_serialized_fields()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/airflow/airflow/decorators/base.py", line 378, in get_serialized_fields
    return super().get_serialized_fields() | {"partial_op_kwargs"}
TypeError: super(type, obj): obj must be an instance or subtype of type
```

I’m suspecting this has something to do with `@attr.define`.

Reply:
Hmm no this is due to `@cache`.

Reply:
Using the “old style” `super` to work around this.

```python
sup = super(DecoratedMappedOperator, DecoratedMappedOperator)
return sup.get_serialized_fields() | {"partial_op_kwargs"}
```

Experience with Python 2 is still useful sometimes 🙂